### ISSUE TITLE
Find and Replace crashes when using group back-reference in replacement text

### ISSUE THREAD

**Issue by gene-pavlovsky** [fdorg/flashdevelop#2000]
I have a file containing strings like this:
`storeLayoutViewState(CLayoutViewState.OFFLINE);`

I use "Find and Replace", in Regular expressions mode.
Find what is: `storeLayoutViewState\(([^)]*)\);`
Replace with is: `localStorageModel.layoutViewState = $1;`

If I use "Find Next" button, the lines I expected to match are found correctly.
If I press the "Replace" button, FD crashes.

I have the latest development version running under Visual Studio debugger, when I press the "Replace" button, this exception is thrown, FD is then terminated:
```
System.NullReferenceException occurred
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=<Cannot evaluate the exception source>
  StackTrace:
   at PluginCore.FRService.FRSearch.ExpandGroups(String text, SearchMatch match) in C:\project\flashdevelop\PluginCore\PluginCore\FRService\FRSearch.cs:line 102
```
Basically, the `match` param the function is getting is `null`.

**Comment by gene-pavlovsky** [comment]
Strangely, if I set a breakpoint at `void ReplaceButtonClick`, the code flow goes normally and no error happens.

**Comment by SlavaRa** [comment]
I can fix it at the weekend(maybe), if no one does it before.

**Comment by gene-pavlovsky** [comment]
Do you have any idea about the possible causes? It was reproducible several times (FD crashed, I ran it again 3 o 4 times), but when I tried to debug it, the problem went away, and I can't reproduce it again.

**Comment by SlavaRa** [comment]
#1183

**Comment by gene-pavlovsky** [comment]
Most likely caused by #1183, I will close this and make a new one if the problem persists
